### PR TITLE
Add ESLint checks

### PR DESCRIPTION
**Pull request by pascal-zarrad**
## Description  
Add TSLint and a gulp task to check code quality. 

## Related issue  
#33 

## How has this been tested?  

- Run `pnpm run lint`. It should run TSLint.

- Violate code style in `src/server.ts` (like removing a semicolon). Run `pnpm run lint`. TSLint should complain about the missing semicolon.

## Motivation and context  
We need to ensure code style and quality.

### CODE REVIEW

**Comment by pascal-zarrad** [comment]
@nicojust @flaglias Migrated to ESLint. PR is now ready for review. The test process is the same as before.
The checks got a bit weaker, but this shouldn't be a big deal.

File: package.json [hunk 1]
```diff
@@ -30,10 +31,19 @@
...
+    "eslint-plugin-import": "^2.18.2",
+    "eslint-plugin-prefer-arrow": "^1.1.6",
     "gulp": "^4.0.2",
     "gulp-clean": "^0.4.0",
+    "gulp-eslint": "^6.0.0",
     "gulp-typescript": "^5.0.1",
     "tslint": "^5.20.0",
-    "typescript": "^3.6.4"
+    "typescript": "^3.6.4",
+    "typescript-eslint": "^0.0.1-alpha.0"
```
Comment (nicojust) [inline package.json]: Is there a stable version available?

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        "@typescript-eslint/no-namespace": "error",
+        "@typescript-eslint/no-parameter-properties": "off",
+        "@typescript-eslint/no-use-before-declare": "off",
+        "@typescript-eslint/no-var-requires": "error",
+        "@typescript-eslint/prefer-for-of": "error",
+        "@typescript-eslint/prefer-function-type": "error",
+        "@typescript-eslint/prefer-namespace-keyword": "error",
+        "@typescript-eslint/quotes": [
+            "error",
+            "double",
```
Comment (nicojust) [inline .eslintrc.js]: I agree with the usage of double quotes. What do you think @flaglias?

Some sources on this topic:
- https://staxmanade.com/2018/03/should-i-use-javascript-single-or-double-quotes/#targetText=There%20is%20really%20no%20difference,other%20character%20within%20a%20string.
- https://stackoverflow.com/a/242833
- https://www.reddit.com/r/javascript/comments/4m715v/should_i_use_or/d3t10sq/

Comment (flaglias) [reply .eslintrc.js]: double quotes are good 👍

**Pull request merged**